src/main/java/org/dependencytrack/policy/PolicyEngine.java

```diff
@@ -150,4 +150,17 @@ private boolean isPolicyAssignedToProjectTag(Policy policy, Project project) {
         }
         return false;
     }
+
+    private boolean isPolicyAssignedToParentProject(Policy policy, Project child) {
+        boolean isParentProjectAssigned = false;
+        if (child.getParent() == null) {
+            return false;
+        }
+        if (isParentProjectAssigned || policy.getProjects().stream().anyMatch(p -> p.getId() == child.getParent().getId())) {
+            return true;
+        } else {
+            isParentProjectAssigned = isPolicyAssignedToParentProject(policy, child.getParent());
+        }
+        return isParentProjectAssigned;
```

Comment:
Could be simplified, or?

```suggestion
        if (child.getParent() == null) {
            return false;
        }
        if (policy.getProjects().stream().anyMatch(p -> p.getId() == child.getParent().getId())) {
            return true;
        } 
        return isPolicyAssignedToParentProject(policy, child.getParent());
```